data-structures-csharp/data-structures-csharp/CircularBuffer/CircularBuffer.cs

```diff
@@ -32,52 +33,81 @@ private void ObjectInvariant()
             Contract.Invariant(Count <= Capacity);
         }
 
-        public CircularBuffer() 
-        {
-            internalList = new List<T>();
-        }
-
+        /// <summary>
+        /// Instantiate a new circular buffer.
+        /// </summary>
+        /// <param name="capacity">The maximum size of the buffer before it begins overwriting itself.</param>
         public CircularBuffer(int capacity)
         {
             Contract.Requires<ArgumentOutOfRangeException>(capacity > 0);
-
-            internalList = new List<T>(capacity);
+            buffer = new T[capacity];
+            Capacity = capacity;
+            Count = 0;
+            addIndex = 0;
+            getIndex = 0;
         }
 
+        /// <summary>
+        /// Add an element to the circular buffer.
+        /// </summary>
         public void Add(T item)
         {
-            throw new NotImplementedException();
+            Contract.Requires<ArgumentNullException>(item != null);
+            buffer[addIndex] = item;
+
+            if (Count < Capacity)
+                Count++;
+
```

Comment:
This is a bit iffy to me since circular buffers wrap around on themselves. I simply saw the Count as being the number of cells in the buffer that were presently full. What would you do in that case?